dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -160,68 +125,14 @@ cdef class cyDiscreteQuadraticModel:
             if irow[qi] == icol[qi]:
                 raise ValueError("quadratic data contains a self-loop")
 
-        cdef cyDiscreteQuadraticModel dqm = cls()
-
-        # set the BQM
-        if num_interactions:
-            dqm.bqm_ = cppAdjVectorBQM[CaseIndex, Bias](
-                &irow[0], &icol[0], &quadratic_biases[0],
-                num_interactions, True)
-
-        # add the linear biases
-        while dqm.bqm_.num_variables() < num_cases:
-            dqm.bqm_.add_variable()
-        for ci in range(num_cases):
-            dqm.bqm_.set_linear(ci, linear_biases[ci])
-
-        # set the case starts
-        dqm.case_starts_.resize(case_starts.shape[0] + 1)
-        for v in range(case_starts.shape[0]):
-            dqm.case_starts_[v] = case_starts[v]
-        dqm.case_starts_[case_starts.shape[0]] = dqm.bqm_.num_variables()
-
-        # and finally the adj. This is not really the memory bottleneck so
-        # we can build an intermediate (unordered) set version
-        cdef vector[unordered_set[VarIndex]] adjset
-        adjset.resize(num_variables)
-        u = 0
-        for ci in range(dqm.bqm_.num_variables()):
-            
-            # we've been careful so don't need ui < case_starts.size() - 1
-            while ci >= dqm.case_starts_[u+1]:
-                u += 1
-
-            span = dqm.bqm_.neighborhood(ci)
-
-            v = 0
-            while span.first != span.second:
-                cj = deref(span.first).first
-
-                # see above note
-                while cj >= dqm.case_starts_[v+1]:
-                    v += 1
+        cdef cyDiscreteQuadraticModel cyDQM = cls()
```

Comment:
Quite confused regarding the next 5 lines of code. I will look up more into this or talk to you in person.